L1Trigger/TrackFindingTMTT/src/MiniHTstage.cc

```diff
@@ -0,0 +1,272 @@
+        numPhiSecPerNon_(settings_->numPhiSectors() / numPhiNonants_),
+        numEtaRegions_(settings_->numEtaRegions()),
+        busySectorKill_(settings_->busySectorKill()),
+        busySectorNumStubs_(settings_->busySectorNumStubs()),
+        busySectorMbinRanges_(settings_->busySectorMbinRanges()),
+        chosenRofPhi_(settings_->chosenRofPhi()),
+        // Get size of 1st stage HT cells.
+        binSizeQoverPtAxis_(miniHoughNbinsPt_ * 2. / (float)settings->houghMinPt() / (float)houghNbinsPt_),
+        binSizePhiTrkAxis_(miniHoughNbinsPhi_ * 2. * M_PI / (float)settings->numPhiSectors() / (float)houghNbinsPhi_),
+        invPtToDphi_(settings_->invPtToDphi()),
+        nHTlinksPerNonant_(0) {
+    nMiniHTcells_ = miniHoughNbinsPt_ * miniHoughNbinsPhi_;
+
+    if (miniHoughLoadBalance_ > 0) {
+      if (muxOutputsHT_ == 3) {
+        nHTlinksPerNonant_ = busySectorMbinRanges_.size() - 1;
+      } else if (muxOutputsHT_ == 2) {
+        nHTlinksPerNonant_ = (busySectorMbinRanges_.size() - 1) * numPhiSecPerNon_;
+      } else {
+        throw cms::Exception(
+            "MiniHTState: hard-wired MHT load balancing can't be enabled as assumes HT output MUX scheme >= 2");
+      }
+    }
+  }
+
+  void MiniHTstage::exec(matrix<HTrphi>& mHtRphis) {
+    for (unsigned int iPhiNon = 0; iPhiNon < numPhiNonants_; iPhiNon++) {
+      map<pair<unsigned int, unsigned int>, unsigned int>
+          numStubsPerLinkStage1;  // Indices are ([link ID, MHT cell], #stubs).
+      map<pair<unsigned int, unsigned int>, unsigned int>
+          numStubsPerLinkStage2;                        // Indices are ([link ID, MHT cell], #stubs).
+      map<unsigned int, unsigned int> numStubsPerLink;  // Indices are ([link ID, MHT cell], #stubs).
+      for (unsigned int iSecInNon = 0; iSecInNon < numPhiSecPerNon_; iSecInNon++) {
+        unsigned int iPhiSec = iPhiNon * numPhiSecPerNon_ + iSecInNon;
+        for (unsigned int iEtaReg = 0; iEtaReg < numEtaRegions_; iEtaReg++) {
+          Sector sector;
+          sector.init(settings_, iPhiSec, iEtaReg);
+          const float& phiCentre = sector.phiCentre();
+          HTrphi& htRphi = mHtRphis(iPhiSec, iEtaReg);
+          const vector<L1track2D>& roughTracks = htRphi.trackCands2D();
+          vector<L1track2D> fineTracks;
+
+          for (const L1track2D& roughTrk : roughTracks) {
+            float roughTrkPhi =
+                reco::deltaPhi(roughTrk.phi0() - chosenRofPhi_ * invPtToDphi_ * roughTrk.qOverPt() - phiCentre, 0.);
+            const pair<unsigned int, unsigned int>& cell = roughTrk.getCellLocationHT();
+            const vector<const Stub*>& stubs = roughTrk.getStubs();
+            bool fineTrksFound = false;
+            bool storeCoarseTrack = false;
+            const unsigned int& link = roughTrk.optoLinkID();
+
+            if (fabs(roughTrk.qOverPt()) <
+                1. / miniHoughMinPt_) {  // Not worth using mini-HT at low Pt due to scattering.
```

Comment:
This would only occur if the python cfg file specified that tracking should be done down to zero Pt, and I'd hope people would realize that that is impossible ...

Reply:
I would still recommend an explicit check for this (maybe even when this parameter is obtained from the cfg)